src/interfaces/chain.h

```diff
@@ -16,17 +15,39 @@
 #include <vector>
 
 class CBlock;
+class CFeeRate;
 class CScheduler;
 class CValidationState;
 class uint256;
+enum class RBFTransactionState;
 struct CBlockLocator;
 struct FeeCalculation;
 
 namespace interfaces {
 
 class Wallet;
 
-//! Interface for giving wallet processes access to blockchain state.
+//! Interface giving clients (wallet processes, maybe other analysis tools in
+//! the future) ability to access to the chain state, receive notifications,
+//! estimate fees, and submit transactions.
+//!
+//! TODO: Current chain methods are too low level, exposing too much of the
```

Comment:
nit: I suspect this and below won't be particularly helpful if included in the doxygen output

Reply:
I think it is helpful to describe the long term goal better (and give ideas to work on for new contributors)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15531#discussion_r262466357

> nit: I suspect this and below won't be particularly helpful if included in the doxygen output 

I don't know if this is a formatting suggestion or a content suggestion. It would help if you could be more specific. I've seen links posted to bitcoin doxygen output once in a while, but I've haven't used the doxygen output myself. I do read comments that appear in code. And I especially like comments that tell me why a particular piece of code was designed a certain way, so I am not left in the dark about things that seem like counterintuitive decision designs.